### ISSUE TITLE
ubuntu kernal update, ptero panel and new grub profile, wrong spark cpu indicators

### ISSUE THREAD

**Issue by LucidAPs** [lucko/spark#192]
After a new kernal update on ubuntu, the ram and cpu statistics for the ptero panel stopped working, so we had to switch profiles in grub, in order to make them work. This had a side effect of completely messing up sparks cpu indicators.

in game
![image](https://user-images.githubusercontent.com/42933023/159452898-13c90e5a-3ef0-46e2-8d2a-440f440bccc0.png)
putty
![image](https://user-images.githubusercontent.com/42933023/159453120-b2fadfcd-62ee-4782-a868-10bd2c3c25c0.png)

You can read more on this issue here.
https://github.com/pterodactyl/panel/issues/3999

**Comment by LucidAPs** [comment]
> ![image](https://user-images.githubusercontent.com/21240809/159492469-7ea57747-71a8-4ec6-b97a-cfa40333a956.png) ![image](https://user-images.githubusercontent.com/21240809/159492652-b60cc996-7da9-4b3a-a2c5-591b43415bb3.png)
> 
> I was getting the same values before making this group setting. spark is showing my correct values i'm not sure

You are getting the correct values before and after changing crub? Did you reboot after changing crub? Are you using ubuntu?

**Comment by LucidAPs** [comment]
I am on Ubuntu 20.04lts maybe that's the difference.

**Comment by lucko** [comment]
I've written a page in the docs which will hopefully be useful: https://spark.lucko.me/docs/misc/About-spark-metrics

tl;dr is that spark gets all of it's metrics data either from the server itself (e.g. the Bukkit API) or from methods in the Java standard library. If the raw data is wrong, unfortunately there's not a whole lot I can do about it.

**Comment by LucidAPs** [comment]
> I've written a page in the docs which will hopefully be useful: https://spark.lucko.me/docs/misc/About-spark-metrics
> 
> tl;dr is that spark gets all of it's metrics data either from the server itself (e.g. the Bukkit API) or from methods in the Java standard library. If the raw data is wrong, unfortunately there's not a whole lot I can do about it.
> 
> I think you guys realise that already, and I appreciate you reporting it nonetheless, but I thought it would be good to clarify. :)

I see, maybe the ptero server image might need updating then to adjust to these changes.

**Comment by Sxtanna** [comment]
relevant? https://github.com/pterodactyl/panel/issues/3988

**Comment by LucidAPs** [comment]
> relevant? [pterodactyl/panel#3988](https://github.com/pterodactyl/panel/issues/3988)

Yeap, good work finding this sir. This is related indeed.

**Comment by lucko** [comment]
Closing as this is a pterodactyl bug.
